doc/source/reference/random/extending.rst

```diff
@@ -99,7 +111,23 @@ removing bounds checks and wrap around, providing array alignment information
         random_values = np.empty(n)
         # Best practice is to release GIL and acquire the lock
         with x.lock, nogil:
-            for i in range(n):
+            for i in range((n+1)//2):
+                x1 = 2.0*rng. - 1.0
+                x2 = 2.0*next_f(state) - 1.0
+                r2 = x1*x1 + x2*x2
+                while r2 >= 1.0 or r2 == 0.0:
+                    x1 = 2.0*next_f(state) - 1.0
+                    x2 = 2.0*next_f(state) - 1.0
+                    r2 = x1*x1 + x2*x2
+                g = np.sqrt(-2.0*np.log(r2)/r2)
+                random_values[2*i] = g*x1
+                if 2*i+1 < n:
+                    random_values[2*i+1] = g*x2
+        randoms = np.asarray(random_values)
+        return randoms
+
+    def normals_zig(Py_ssize_t n):
+
                 random_values[i] = random_gauss_zig(rng)
```

Comment:
the `_zig` is no longer in the name

Reply:
this whole block should be included via a `.. literalinclude` directive from the appropriate example file